ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -267,6 +293,29 @@ private CloudBlobContainer getContainer(BlobId blobId, boolean autoCreate)
     return azureContainer;
   }
 
+  /**
+   * Get the blob name to use in Azure Blob Storage
+   * @param blobId The {@link BlobId} to store.
+   * @return An Azure-friendly blob name.
+   */
+  private String getAzureBlobName(BlobId blobId) {
+    // Prefix to assist in blob data sharding, since beginning of blobId has little variation.
+    String prefix = blobId.getUuid().substring(0,4) + "-";
```

Comment:
```return blobId.getUuid().substring(0,4) + "-" + blobId.getID()```?

Reply:
I prefer to make the prefix explicit and documented.